gradle/publish.gradle

```diff
@@ -1,9 +1,102 @@
+version = '0.3.1'
+String tag = "v$project.version"
+groovydoc.docTitle = 'Static Analysis Plugin'
+
 apply plugin: 'com.novoda.bintray-release'
 publish {
     userOrg = 'novoda'
     groupId = 'com.novoda'
     artifactId = 'gradle-static-analysis-plugin'
-    publishVersion = '0.3'
+    publishVersion = project.version
     website = 'https://github.com/novoda/gradle-static-analysis-plugin'
 }
 
+apply plugin: 'com.novoda.build-properties'
+buildProperties {
+    secrets {
+        file(rootProject.file('secrets.properties'), '''
+This file should contain:
+- git.username: the username used to push to the repo
+- git.password: the password used to push to the repo
+		''')
+    }
+}
+
+apply plugin: 'org.ajoberstar.grgit'
+apply plugin: 'org.ajoberstar.github-pages'
+
+githubPages {
+    commitMessage = "Deploy groovydoc for release $tag"
+    pages {
+        from groovydoc.destinationDir
+        into "docs/${project.version}"
+    }
+}
+
+task prepareGhCredentials {
```

Comment:
yeah, that extension (and the incosistent naming) is coming from the aforementioned plugin 👼